### PR TITLE
Fix: Prevent tick-related errors in listings & bidding

### PR DESCRIPTION
**Pull request by sjc5**
## Problem:

- Lack of form validation is causing unnecessary tick-related errors

## This Fix:

- Lowers the permitted precision (decimal increment) of the price and tick size inputs to match bidding precision
- Adds form validation to make sure the price is an exact multiple of the tick size when creating a listing (and throws an error / disables the submit button if not)

### CODE REVIEW

File: js/packages/web/src/views/auctionCreate/priceAuction.tsx [hunk 1]
```diff
@@ -16,72 +40,88 @@ export const PriceAuction = (props: {
...
-        <h3>Tick Size</h3>
+        <h3 className="metaplex-margin-top-4">Tick Size</h3>
         <p>All bids must fall within this price increment.</p>
-        <Input
-          type="number"
+        <InputNumber<number>
           min={0}
+          decimalSeparator="."
+          className="metaplex-fullwidth"
+          step="0.01"
```
Comment (kespinola) [inline js/packages/web/src/views/auctionCreate/priceAuction.tsx:99]: Can the validation for tick size be done as a validation function attached to the input or do we not have this part of the fields wrapped in a form yet? Using a form here like we did with the nft descriptive fields is a good call.

**Review by kespinola — CHANGES_REQUESTED** [review]

File: js/packages/web/src/views/auctionCreate/priceAuction.tsx [hunk 2]
```diff
@@ -1,90 +1,173 @@
...
 import { AuctionCategory, AuctionState } from '.';
 
 export const PriceAuction = (props: {
   attributes: AuctionState;
   setAttributes: (attr: AuctionState) => void;
   confirm: () => void;
 }) => {
+  type InputState = number | null;
+  const [price, setPrice] = useState<InputState>(null);
+  const [tickSize, setTickSize] = useState<InputState>(null);
```
Comment (kespinola) [inline js/packages/web/src/views/auctionCreate/priceAuction.tsx:13]: Can we drop the the price and tickSize as local state? Its tracked in a attributes yes?

**Review by kespinola — CHANGES_REQUESTED** [review]

File: js/packages/web/src/views/auctionCreate/priceAuction.tsx (hunk 2 above)
Comment (sjc5) [reply js/packages/web/src/views/auctionCreate/priceAuction.tsx]: Yeah that's cleaner.  Resubmitted, dropping the local state

**Review by kespinola — APPROVED** [review]
This is awesome. TY.

**Pull request merged**